### PR TITLE
Update so dependents can compile

### PR DESCRIPTION
**Pull request by olix0r**
Projects cannot currently depend on the h2 repo due the following compile errors:

```
error: method is never used: `new`
   --> /Users/ver/b/rs/h2/src/frame/headers.rs:265:5
    |
265 | /     pub fn new(
266 | |         stream_id: StreamId,
267 | |         promised_id: StreamId,
268 | |         pseudo: Pseudo,
...   |
279 | |         }
280 | |     }
    | |_____^
    |
note: lint level defined here
   --> /Users/ver/b/rs/h2/src/lib.rs:1:9
    |
1   | #![deny(warnings, missing_debug_implementations)]
    |         ^^^^^^^^
    = note: #[deny(dead_code)] implied by #[deny(warnings)]

error: method is never used: `into_parts`
   --> /Users/ver/b/rs/h2/src/frame/headers.rs:339:5
    |
339 | /     pub fn into_parts(self) -> (Pseudo, HeaderMap) {
340 | |         (self.header_block.pseudo, self.header_block.fields)
341 | |     }
    | |_____^

error: method is never used: `fields`
   --> /Users/ver/b/rs/h2/src/frame/headers.rs:343:5
    |
343 | /     pub fn fields(&self) -> &HeaderMap {
344 | |         &self.header_block.fields
345 | |     }
    | |_____^

error: method is never used: `into_fields`
   --> /Users/ver/b/rs/h2/src/frame/headers.rs:347:5
    |
347 | /     pub fn into_fields(self) -> HeaderMap {
348 | |         self.header_block.fields
349 | |     }
    | |_____^

error: method is never used: `set_max_concurrent_streams`
  --> /Users/ver/b/rs/h2/src/frame/settings.rs:77:5
   |
77 | /     pub fn set_max_concurrent_streams(&mut self, max: Option<u32>) {
78 | |         self.max_concurrent_streams = max;
79 | |     }
   | |_____^

error: aborting due to 5 previous errors

error: Could not compile `h2`.

To learn more, run the command again with --verbose.
```

This change adds two `allow(dead_code)` annotations to prevent these compile errors.

Furthermore, `cargo fmt` has been run.

### CODE REVIEW

**Comment by carllerche** [comment]
It looks like this is also running a `cargo fmt`. Same request as in #112 could we make a standalone format PR and then rebase these PRs on that?

File: src/frame/headers.rs [hunk 1]
```diff
@@ -261,6 +261,7 @@ impl fmt::Debug for Headers {
 
 // ===== impl PushPromise =====
 
+#[allow(dead_code)]
```
Comment (carllerche) [inline src/frame/headers.rs]: This should actually be flagged w/ unstable I think. I'll update the PR.

**Comment by seanmonstar** [comment]
Any errors from lints in dependencies should never fail compilation, cargo caps lints to warnings. If this isn't the case, that sounds like a bug in cargo or something.

**Comment by carllerche** [comment]
I updated the PR, so I probably shouldn't approve it?

**Comment by carllerche** [comment]
@seanmonstar I'm guessing it is failing because it is a git dep.

**Comment by carllerche** [comment]
Ugh, I think this isn't quite right.

**Comment by carllerche** [comment]
Ok, I think this should fix it.

**Pull request merged**